src/peer_manager.rs

```diff
@@ -1215,21 +1213,27 @@ impl PeerManager {
         let _ = self.set_state(peer_id, PeerState::Routing(RoutingConnection::Direct));
     }
 
-    /// Returns the `NUM_TUNNEL_VIA_NODES` closest peers from our section which can be potential
-    /// tunnel node.
-    pub fn potential_tunnel_nodes(&self) -> Vec<(XorName, PeerId)> {
-        let our_section = self.routing_table
-            .other_close_names(self.our_public_id.name())
-            .unwrap_or_default();
-        self.peer_map
-            .peers()
-            .filter_map(|peer| if our_section.contains(peer.name()) &&
-                                  peer.state.can_tunnel_for() {
-                peer.peer_id.map_or(None, |peer_id| Some((*peer.name(), peer_id)))
-            } else {
-                None
+    /// Returns the closest section to the peer and the closest section to own.
+    pub fn potential_tunnel_nodes(&self, name: &XorName) -> Vec<(XorName, PeerId)> {
+        self.routing_table
+            .our_section()
+            .iter()
+            .chain(self.routing_table.closest_section(name).1.iter())
```

Comment:
@maqi if it returns None, that should be fine right? cos that means we dont have anyone else "yet" from the target section, which means we'll soon be having some people added and at that point we'll query them if they can tunnel to this person anyways from the add_to_routing_table msg flow. This here is going to get us someone who we may loose a lot earlier due to further future splits and that can be prevented by using the other option maybe.